doc/releases/changelog-dev.md

```diff
@@ -93,6 +93,9 @@
   to their top-level pages (rather than their module-level pages, eg. `qml.measurements.expval`).
   [(#4750)](https://github.com/PennyLaneAI/pennylane/pull/4750)
 
+* The mentioning of QNodecollection has been removed from the Installation and dependencies page because 
+  the `collections` module has been removed from PennyLane since v0.31.
+
```

Comment:
```suggestion
* The mentioning of QNodecollection has been removed from the Installation and dependencies page because 
  the `collections` module has been removed from PennyLane since v0.31.
  [(#4857)](https://github.com/PennyLaneAI/pennylane/pull/4857)
```